core/domain/question_services.py

```diff
@@ -323,10 +334,11 @@ def get_question_summaries_linked_to_skills(
         a time is not supported currently.
 
     Returns:
-        list(QuestionSummary), str|None. The list of question summaries linked
-            to the given skill_ids and the next cursor value to be used for the
-            next page (or None if no more pages are left). The returned next
-            cursor value is urlsafe.
+        list(dict), str|None. The list of question summaries and the
```

Comment:
Should return domain object.

Reply:
The object returned here would be directly used by controller as response, so should I create a separate domain object for just this function? 
Since, such a domain object is not used anywhere else and is exclusive for this function. Similarly for `get_questions_and_skill_descriptions_by_skill_ids`

Reply:
I think your latter approach sounds better. In general pass domain objects around. Dicts should only be used at the application boundaries (i.e. the controller interface to the outside world).

Reply:
Ok, done.